### PR TITLE
Flask lesson advanced

### PR DESCRIPTION
**Pull request by AlexandrVovk**
Flask lesson advanced

### CODE REVIEW

File: flask_lesson_advanced/blueprint/product_main.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+def products_fun():
+    arg_price = request.args.get("price")
+    products_list = [{"name": rest.name, "id": rest.id, "price": rest.price,
+                      "img_name": rest.img_name,
+                      "description": rest.description}
+                     for rest in DB['products']]
+    product_filter = [i for i in products_list if str(i["price"]) == str(arg_price)]
+    sess = session.keys()
+    if len(product_filter) > 0:
+        return render_template('all_products.html', value=product_filter, sess=sess)
```
Comment (bkalika) [inline flask_lesson_advanced/blueprint/product_main.py]: для чого тут 2 однакових ретурна?

Comment (AlexandrVovk) [reply flask_lesson_advanced/blueprint/product_main.py]: > для чого тут 2 однакових ретурна?

вони різні, а саме значення value=
        return render_template('all_products.html', value=product_filter, sess=sess)
    return render_template('all_products.html', value=products_list, sess=sess)

File: flask_lesson_advanced/blueprint/product_main.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                    static_folder='static_product')
+
+
+@product.route('/product', methods=['GET'])
+def products_fun():
+    arg_price = request.args.get("price")
+    products_list = [{"name": rest.name, "id": rest.id, "price": rest.price,
+                      "img_name": rest.img_name,
+                      "description": rest.description}
+                     for rest in DB['products']]
```
Comment (dimigor) [inline flask_lesson_advanced/blueprint/product_main.py:17]: move this on if statement bellow

File: flask_lesson_advanced/blueprint/product_main.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    return render_template('all_products.html', value=products_list, sess=sess)
+
+
+@product.route('/product/<id>', methods=['GET'])
+def product_id_fun(id):
+    products_list = [{"name": rest.name, "id": rest.id, "price": rest.price,
+                      "img_name": rest.img_name,
+                      "description": rest.description}
+                     for rest in DB['products']]
+    for product_dict in products_list:
```
Comment (dimigor) [inline flask_lesson_advanced/blueprint/product_main.py:35]: what if you cannot find any product?
return 404 pag at `for...else`

File: flask_lesson_advanced/blueprint/supermarket_main.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    supermarket_filter = [i for i in supermarkets_list if str(i["location"]) == str(arg_location)]
+    sess = session.keys()
+    if len(supermarket_filter) > 0:
+        return render_template('all_supermarkets.html', value=supermarket_filter, sess=sess)
+    return render_template('all_supermarkets.html', value=supermarkets_list, sess=sess)
+
+
+@supermarket.route('/supermarket/<id>', methods=['GET'])
+def supermarket_id_fun(id):
+    print("supermarket_id_fun")
```
Comment (dimigor) [inline flask_lesson_advanced/blueprint/supermarket_main.py]: remove print

File: flask_lesson_advanced/blueprint/template_product/add_product.html [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  <br><br>
+  Price:<br>
+  <input name="price">
+  <br><br>
+  Img_name:<br>
+  <input name="img_name" type="file">
+  <br><br>
+  <input type="submit" value="Add Product">
+</form>
+{% endblock %}
```
Comment (dimigor) [inline flask_lesson_advanced/blueprint/template_product/add_product.html:21]: endline

File: flask_lesson_advanced/blueprint/template_supermarket/all_supermarkets.html [hunk 6]
```diff
@@ -0,0 +1,16 @@
+{% extends "base.html" %}
+{% block title %}All Supermarkets{% endblock %}
+{% block body %}
+<h1>Supermarkets</h1>
+    <ol>
+        {% for i in value %}
+        <li>
+            {% if i['name'] in sess %}
+                <a href="{{url_for('supermarket.supermarkets_fun')}}/{{ i['id'] }}" onclick="return false;">{{ i['name'] }}</a>
```
Comment (dimigor) [inline flask_lesson_advanced/blueprint/template_supermarket/all_supermarkets.html]: `url_for` has parameter for `/{{ i['id'] }}`

File: flask_lesson_advanced/templates/home.html [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    <ul>
+        <li>
+            {% if value == True %}
+                <a href="/product" onclick="return false;">Products</a>
+            {% else %}
+                <a href="/product" >Products</a>
+            {% endif %}
+        </li>
+
+        <li><a href="/supermarket">Supermarkets</a></li>
```
Comment (dimigor) [inline flask_lesson_advanced/templates/home.html]: use `url_for` instead

**Review by dimigor — CHANGES_REQUESTED** [review]

File: flask_lesson_advanced/blueprint/product_main.py (hunk 3 above)
Comment (AlexandrVovk) [reply flask_lesson_advanced/blueprint/product_main.py:35]: done

File: flask_lesson_advanced/blueprint/supermarket_main.py (hunk 4 above)
Comment (AlexandrVovk) [reply flask_lesson_advanced/blueprint/supermarket_main.py]: done

File: flask_lesson_advanced/blueprint/template_product/add_product.html (hunk 5 above)
Comment (AlexandrVovk) [reply flask_lesson_advanced/blueprint/template_product/add_product.html:21]: done

File: flask_lesson_advanced/blueprint/product_main.py (hunk 2 above)
Comment (AlexandrVovk) [reply flask_lesson_advanced/blueprint/product_main.py:17]: we had chat about it and it no need to be changed

File: flask_lesson_advanced/blueprint/product_main.py (hunk 3 above)
Comment (AlexandrVovk) [reply flask_lesson_advanced/blueprint/product_main.py:35]: we had chat about it and I have to make a page with information message

File: flask_lesson_advanced/blueprint/template_supermarket/all_supermarkets.html (hunk 6 above)
Comment (AlexandrVovk) [reply flask_lesson_advanced/blueprint/template_supermarket/all_supermarkets.html]: we had chat about it and it no need to be changed

File: flask_lesson_advanced/templates/home.html (hunk 7 above)
Comment (AlexandrVovk) [reply flask_lesson_advanced/templates/home.html]: done

**Review by dimigor — APPROVED** [review]

**Pull request merged**